### PR TITLE
DOC: update and clean up install instructions

### PR DESCRIPTION
**Pull request by rgommers**
- Use Python 3.5 by default.
- Remove Gentoo, outdated and not big enough to maintain.
- Explain that for pip instructions to work, Python and pip need to already be installed.
- some minor cleanups

### CODE REVIEW

File: www/install.rst [hunk 1]
```diff
@@ -126,19 +106,18 @@ of popular package managers you can install.
...
 
 Homebrew
 ~~~~~~~~
 
 At the time of writing (September 2016), `Homebrew <http://brew.sh/>`_ does
 not have the full SciPy stack available (i.e. you cannot do ``brew install
 <formula>`` for everything).  You can install NumPy, SciPy and Matplotlib,
-with ``brew tap homebrew/python; brew install python numpy scipy matplotlib``,
-but we recommend pip installs in preference to the homebrew builds.
+with ``brew tap homebrew/python.
```
Comment (takluyver) [inline www/install.rst]: Missing the closing backticks here.

Comment (rgommers) [inline www/install.rst]: removed a bit too much there, fixed now.

**Comment by Eric89GXL** [comment]
Looks good to me, +1 for merge

File: www/install.rst [hunk 2]
```diff
@@ -29,34 +29,28 @@ Installing via pip
...
-    python3 -m pip install --upgrade pip
+To install via pip on Mac or Linux, first upgrade pip to the latest version::
 
-Replace ``python3`` with ``python2`` for Python 2 pip upgrade.
+    python -m pip install --upgrade pip
 
 Then install the SciPy stack packages with pip.  We recommend a *user*
 install, using the ``--user`` flag to pip.  This installs packages for your
 local user, and does not need extra permissions to write to the system
 directories::
```
Comment (ev-br) [inline www/install.rst]: I think it'd be worth stressing explicitly that `sudo pip` is *not* recommended. The internet is unfortunately full of recommendations to do just that.

Comment (rgommers) [inline www/install.rst]: agreed. added a note

**Comment by rgommers** [comment]
merging. thanks for the review all

**Pull request merged**